bedevere/bpo.py

```diff
@@ -40,16 +34,21 @@
     issue_number_found = ISSUE_RE.search(event.data["pull_request"]["title"])
     if not issue_number_found:
         issue = await util.issue_for_PR(gh, event.data["pull_request"])
-        status = SKIP_ISSUE_STATUS if util.skip("issue", issue) else FAILURE_STATUS
+        status = (SKIP_ISSUE_STATUS if util.skip("issue", issue)
+                                    else create_failure_status_no_issue())
     else:
+        issue_number = issue_number_found.group("issue")
         if "body" in event.data["pull_request"]:
             body = event.data["pull_request"]["body"] or ""
             if not body or CLOSING_TAG not in body:
-                issue_number = issue_number_found.group("issue")
                 new_body = BODY.format(body=body, issue_number=issue_number)
                 body_data = {"body": new_body, "maintainer_can_modify": True}
                 await gh.patch(event.data["pull_request"]["url"], data=body_data)
```

Comment:
The patching of the first message seems to occur regardless of whether the issue # was validated. Do you think it would be better to not create a URL if we can check that it won't return a 404?

Reply:
Yeah, that sounds like a good improvement.